components/form.js

```diff
@@ -0,0 +1,178 @@
+                        token: { ...data, fetching_time: new Date().getTime() }
+                    })
+                )
+                .catch(error => console.log(error));
+        } else {
+            this.setState({ token: token });
+        }
+    }
+
+    getFormFields() {
+        // Fetch the data form fields from API, use it in state
+        let formFields = JSON.parse(localStorage.getItem('Form Fields'));
+        if (!(Array.isArray(formFields) && formFields.length)) {
+            fetch(
+                baseUrl +
+                    '/api/v3/forms/14/attributes?order=asc&orderby=priority'
+            )
+                .then(response => {
+                    if (response.ok) {
+                        return response.json();
+                    } else {
+                        throw new Error('Field fetching response is not ok.');
+                    }
+                })
+                .then(data =>
+                    this.setState({
+                        formFields: data.results
+                    })
+                )
+                .catch(err => console.log(err));
+        } else {
+            this.setState({
+                formFields: formFields
+            });
+        }
+    }
+
+    componentDidMount() {
+        this.getFormFields();
+        this.getToken();
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        localStorage.setItem(
+            'Form Fields',
+            JSON.stringify(this.state.formFields)
+        );
+        localStorage.setItem('Bearer Token', JSON.stringify(this.state.token));
+    }
+
+    // Handler html For when the submit button is pressed
+    onSubmit = e => {
+        e.preventDefault();
+
+        // Make a post request to Ushahidi sever
+        fetch(baseUrl + '/api/v3/posts', {
+            method: 'POST',
+            body: JSON.stringify(this.state.formData),
+            headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json',
+                Authorization: `Bearer ${this.state.token.access_token}`
+            }
+        })
+            .then(response => {
+                return response.json();
+            })
+            .then(data => {
+                console.log(`Success!!! \n ${data}`);
+            })
+            .catch(error => {
+                console.log(error);
+            });
+
+        this.setState({
+            formData: {}
+        });
+    };
+
+    // Create an array of elements to be rendered in the form
+    FormInputs = props => {
+        return props.fields.map(field => {
+            return (
+                <div key={field.key}>
+                    <label htmlFor={field.label}> {field.label} </label>
+                    <input
+                        type={field.input}
+                        name={field.label}
+                        placeholder={`Enter the ${field.label}`}
+                        value={this.state.formData[field.label]}
```

Comment:
why are you using the field label for the input's value? (just curious) what are we expecting to see here?

Reply:
This is used in the change function for updating the state with the value of the input.